src/app/app-modules/nurse-doctor/workarea/workarea.component.ts

```diff
@@ -4213,6 +4216,59 @@ export class WorkareaComponent
         }
       });
   }
+  getMappedAbdmFacility() {
+    const locationData: any = localStorage.getItem('loginDataResponse');
+    const jsonLoccationData = JSON.parse(locationData);
+    let workLocationId: any;
+    if (jsonLoccationData?.previlegeObj[0]?.roles) {
+      const roles = jsonLoccationData?.previlegeObj[0]?.roles;
+      roles.find((item: any) => {
+        item.RoleName.toLowerCase() === 'doctor';
+        workLocationId = item.workingLocationID;
+      });
+    }
+    console.log('workLocationId', workLocationId);
+    this.registrarService.getMappedFacility(workLocationId).subscribe(
+      (res: any) => {
+        if (res.statusCode === 200 && res.data != null) {
+          const data = res.data;
+          if (data.abdmFacilityID && data.abdmFacilityName) {
```

Comment:
Please add optional chaining or explicit null check for data.abdmFacilityID  , data.abdmFacilityName

Reply:
@devikasuresh20  adding so checks for falsy values which are both null and undefined. so not checking explicity for null condition